### ISSUE TITLE
Kext validation problems in 3.6 beta 12 and 14.

### ISSUE THREAD

**Issue by Thireus** [Tunnelblick/Tunnelblick#286]
The following issue appears when Tunnelblick attempts to load the tap/tun kexts: 
```Can't load /Applications/Tunnelblick.app/Contents/Resources/tap-signed.kext - validation problems.```

Not affected:
Tunnelblick_3.6beta10_build_4400

Affected versions:
Tunnelblick_3.6beta12_build_4435
Tunnelblick_3.6beta14_build_4441

As a result the tun/tap kexts can't be loaded. Downgrading to 3.6 beta 10 fixes the issue.

**Comment by jkbullard** [comment]
If you are *not* using a preview version of OS X 10.11.2, please provide the exact version of OS X you are using (version and build number) and confirm that you installed Tunnelblick by double-clicking the Tunnelblick icon in a downloaded disk image.

If you *are* using a preview version of OS X 10.11.2, then this problem was fixed in commit 7fa1509a41110943a12ed53bf3efa4516554e745 and will be included in the next release. Email me privately at jkbullard at Gmail for a link to a pre-release version of Tunnelblick which includes the fix.

**Comment by Thireus** [comment]
Sorry about that. Yosemite 10.10.5.

**Comment by jkbullard** [comment]
Please post the build number of your copy of OS X. You can get the build number by clicking on the apple icon in the upper left corner of the display, then clicking "About This Mac", then clicking on the "Version 10.10.5" text just below "OS X Yosemite" in the new window. The build number will appear in parentheses immediately after "Version 10.10.5".

**Comment by Thireus** [comment]
I edited the previous comment.

**Comment by jkbullard** [comment]
Are you running as a "standard" user or an "admin" user?

Please [verify the downloaded disk image](https://tunnelblick.net/downloads.html#verifying-downloads) of Tunnelblick 3.6beta14, reinstall Tunnelblick by double-clicking the Tunnelblick icon in that disk image, and try again.

If the problem persists, email me privately at jkbullard as described above.

**Comment by Thireus** [comment]
I can ensure you the dmg images have not been altered. I am running as an admin. Each time I reinstall by double-clicking the Tunnelblick icon in that disk image. I've tried about 3 times each release already just to be really sure. Each time it asks for the admin password to replace the necessary files. Which I double check as well.

```openssl sha1 /Users/thireus/Downloads/Tunnelblick_3.6beta14_build_4441.dmg 
SHA1(/Users/thireus/Downloads/Tunnelblick_3.6beta14_build_4441.dmg)= 1b9a260e732a136c09acbcdac1e7d4c5f53234fa

openssl md5 /Users/thireus/Downloads/Tunnelblick_3.6beta14_build_4441.dmg 
MD5(/Users/thireus/Downloads/Tunnelblick_3.6beta14_build_4441.dmg)= 67f3d2ba10c68ead0e34e245745fd489

openssl sha1 /Users/thireus/Downloads/Tunnelblick_3.6beta12_build_4435.dmg 
SHA1(/Users/thireus/Downloads/Tunnelblick_3.6beta12_build_4435.dmg)= 763103350b633a3a153e3044984254980b8472b6

openssl md5 /Users/thireus/Downloads/Tunnelblick_3.6beta12_build_4435.dmg 
MD5(/Users/thireus/Downloads/Tunnelblick_3.6beta12_build_4435.dmg)= 6307a5883f75acb47db69ce90cd48442```

**Comment by Thireus** [comment]
By the way, if we attempt to patch the kext directly the OS X quarantine system yells about it and prevents the app to be launched. And even if removed from quarantine, your app checks the signature of the kext and warns that it was altered. So trust me, that's not the issue.

**Comment by jkbullard** [comment]
Email me privately at jkbullard at Gmail for a link to a pre-release version of Tunnelblick which includes the fix developed for preview versions of OS X 10.11.2. I suspect it will also resolve your problem.

**Comment by caarlos0** [comment]
@jkbullard I might be having the same issue. Tunnelblick just wont freezes while opening on OSX 10.11.2. I don't know what logs I should look at, but I think it might be related...

Can you send me the developed preview for 10.11.2 too? My email is caarlos0 at gmail dot com.

Thanks

**Comment by jkbullard** [comment]
Tunnelblick 3.6beta16 is available on the [Downloads](https://tunnelblick.net/downloads.html) page and includes the fix.

**Comment by caarlos0** [comment]
Hi @jkbullard , it still not working here:

![image](https://cloud.githubusercontent.com/assets/245435/11827776/fa5600cc-a375-11e5-82e8-96231fe56598.png)

I already uninstalled and installed it again. Any logs I can gather to help debugging this?

**Comment by jkbullard** [comment]
This was fixed in Tunnelblick 3.6beta16.
